src/udev/udev-builtin-net_id.c

```diff
@@ -282,28 +342,35 @@ static int dev_pci_slot(struct udev_device *dev, struct netnames *names) {
                 goto out;
         }
 
-        FOREACH_DIRENT_ALL(dent, dir, break) {
-                int i;
-                char *rest, *address, str[PATH_MAX];
-
-                if (dent->d_name[0] == '.')
-                        continue;
-                i = strtol(dent->d_name, &rest, 10);
-                if (rest[0] != '\0')
-                        continue;
-                if (i < 1)
-                        continue;
-
-                xsprintf(str, "%s/%s/address", slots, dent->d_name);
-                if (read_one_line_file(str, &address) >= 0) {
-                        /* match slot address with device by stripping the function */
-                        if (strneq(address, udev_device_get_sysname(names->pcidev), strlen(address)))
-                                hotplug_slot = i;
-                        free(address);
+        hotplug_slot_dev = names->pcidev;
+        while (hotplug_slot_dev) {
+                FOREACH_DIRENT_ALL(dent, dir, break) {
+                        int i;
+                        char *rest, *address, str[PATH_MAX];
+
+                        if (dent->d_name[0] == '.')
+                                continue;
+                        i = strtol(dent->d_name, &rest, 10);
+                        if (rest[0] != '\0')
+                                continue;
+                        if (i < 1)
+                                continue;
+
+                        xsprintf(str, "%s/%s/address", slots, dent->d_name);
+                        if (read_one_line_file(str, &address) >= 0) {
+                                /* match slot address with device by stripping the function */
+                                if (strneq(address, udev_device_get_sysname(hotplug_slot_dev), strlen(address)))
```

Comment:
`startswith()`?

Reply:
Thanks for the additional feedback, and I apologize for the delay.  I made the changes, except for two... those two ("use safe_atoi()" and one of the "startswith()?") are code that isn't mine... I made changes around that code but didn't actually write that code.  If you want me to change those things anyway, I can do so, just let me know, but that code is currently in systemd like that without my patches.